dbms/src/Storages/getStructureOfRemoteTable.cpp

```diff
@@ -31,9 +32,48 @@ ColumnsDescription getStructureOfRemoteTable(
     const Context & context,
     const ASTPtr & table_func_ptr)
 {
-    /// Send to the first any remote shard.
-    const auto & shard_info = cluster.getAnyShardInfo();
+    const auto & shards_info = cluster.getShardsInfo();
 
+    std::string fail_messages;
+
+    for (auto & shard_info : shards_info)
```

Comment:
Maybe we just need to wrap all shard pools in one ConnectionPoolWithFailover when `skip_unavailable_shards = 1` ?

Reply:
This could be neat (and we get out of the box prioritization based on error rates, could also implement prioritization of local replicas), but it looks like it would need a bit more refactoring which I'm not ready to do at the moment.